onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1541,6 +1660,307 @@ with the exception that numpy default keepdims to False instead of True.)DOC")
+        ValidateTypeAndShapeForScaleAndZP(ctx, 3, ONNX_NAMESPACE::TensorProto::FLOAT, true);
+        ValidateTypeAndShapeForScaleAndZP(ctx, 4, data_type->tensor_type().elem_type(), true);
+
+        int64_t keep_dims = 1;
+        auto attr_proto = ctx.getAttribute("keepdims");
+        if (attr_proto) {
+          keep_dims = attr_proto->i();
+        }
+
+        auto& input_shape = ctx.getInputType(0)->tensor_type().shape();
+        int64_t input_ndim = input_shape.dim_size();
+        auto output_shape =
+            ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
+        std::vector<int64_t> axes;
+        auto axes_proto = ctx.getAttribute("axes");
+        if (axes_proto)
+          axes.assign(axes_proto->ints().begin(), axes_proto->ints().end());
+
+        for (size_t i = 0; i < axes.size(); ++i) {
+          if (axes[i] < -input_ndim || axes[i] >= input_ndim) {
+            fail_shape_inference(
+                "axis must be in [-rank, rank-1]. input rank was ", input_ndim);
+          }
+          if (axes[i] < 0)
+            axes[i] += input_ndim;
+        }
+        // do we need to handle negative axis?
+        for (int i = 0; i < input_ndim; ++i) {
+          // axes empty means reduce all dim
+          if (!axes.empty() &&
+              std::find(axes.begin(), axes.end(), i) == axes.end()) {
+            auto dim = output_shape->add_dim();
+            dim->CopyFrom(input_shape.dim(i));
+          } else {
+            if (keep_dims == 1) {
+              auto dim = output_shape->add_dim();
+              dim->set_dim_value(1);
+            }
+          }
+        }
+      });
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(MulInteger)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(R"DOC(Performs element-wise binary quantized multiplication (with Numpy-style broadcasting support).
+"This operator supports **multidirectional (i.e., Numpy-style) broadcasting**"
+The output of this op is the int32 accumulated result of the mul operation
+
+```
+C (int32) = (A - A_zero_point) * (B - B_zero_point)
+```
+
+)DOC")
+      .Input(0, "A", "First operand.", "T")
+      .Input(
+          1,
+          "A_zero_point",
+          "Input A zero point. Default value is 0 if it's not specified. It's a scalar, which means a per-tensor/layer quantization.",
+          "T",
+          OpSchema::Optional)
+      .Input(2, "B", "Second operand.", "T")
+      .Input(
+          3,
+          "B_zero_point",
+          "Input B zero point. Default value is 0 if it's not specified. It's a scalar, which means a per-tensor/layer quantization.",
+          "T",
+          OpSchema::Optional)
+      .Output(0, "C", "Constrain output to 32 bit tensor", "T1")
+      .TypeConstraint("T", {"tensor(uint8)", "tensor(int8)"}, "Constrain input types to 8 bit signed and unsigned tensors.")
+      .TypeConstraint("T1", {"tensor(int32)"}, "Constrain output types to 32 bit tensors.")
```

Comment:
(minor) T1 and T2? (consistent with Cast and ConstantOfShape)

https://github.com/onnx/onnx/blob/master/docs/Operators.md#cast
https://github.com/onnx/onnx/blob/master/docs/Operators.md#constantofshape